blog-posts/event-sourcing-serveless/event-sourcing-serveless.md

```diff
@@ -0,0 +1,171 @@
+
+CQRS (Command and Query Responsibility Segregation) is a pattern that separates read and write operations for a data store solution. In our case the credit and debit events are written into a database different from the database which the data is read from.
+
+It's a good thing to know CQRS to better grasp what will be approached in this article, but don't worry it is not required. Numerous resources already deal with this topic and you can easily look it up on the web.
+
+## What we will build together
+
+Let's build the example of the banking application. It needs to display both the balance of your account and the last transaction you made. We will have a much bigger need in reading resources than in writing, therefore CQRS is appropriate for the use case. Moreover, CQRS is the best reading/writing interface for an event sourcing data storing solution. For now let's take a look at what the architecture of such an application should look like.
+
+<img src="./assets/event-sourcing-x-CQRS-simple-schema.png" style="display: block; margin: auto" width="600"/>
+
+If you are user 2, you see that the application will fetch the total balance of your account from a read model that is called a **projection**. The entity that handles that action is called a query (cQrs). The projection of the balance is computed from the events stored in a database called the **event store**. A data synchronization system must be implemented to have accurate reading data compared to the source of truth. And when you make a transaction, a request is made to a **command** (Cqrs) that will write in the event store the event.
+
+Now let's see how to implement it with AWS serverless technologies.
+
+## Event Store
+
+The source of truth of the application is the event store. All credit and debit events for every user are stored in it. We want a quick to access serverless database, therefore DynamoDB is the most appropriate. We have to define the access pattern to retrieve every events of a user in a chronological order. We can store the number of the event in a field named **version** and put it as `sort key` (SK). To easily retrieve data from a specific user we can use the **id of a user** as a `partition key` (PK). That way we have a composite primary key with the PK and the SK.
+
+<img src="./assets/event-store-schema.png" style="display: block; margin: auto" width="600"/>
+
+We start building our application. The event store represents its first block, and it contains credit and debit events of every user.
+
+<img src="./assets/archi-step-1-event-store.png" style="display: block; margin: auto" height="300"/>
+
+> :face_with_monocle: It is nice to be able to store data, but the question now is how do we store them?
+
+## Command
+
+We need to fuel this event store with some events. The command, which is the C of CQRS interface, is in charge of this. It's basically a handler that writes an event into the event store when triggered.
```

Comment:
Fuel ça veut dire "to provide power to something" du coup c'est ça que tu voulais dire ?
(ça veut pas dire "remplir")